march_joint_inertia_controller/src/inertia_estimator.cpp

```diff
@@ -0,0 +1,290 @@
+// Copyright 2020 Project March.
+#include <hardware_interface/joint_command_interface.h>
+#include <hardware_interface/joint_state_interface.h>
+#include "march_joint_inertia_controller/inertia_estimator.h"
+#include "math.h"
+#include "ros/ros.h"
+#include "std_msgs/Float64.h"
+
+void absolute(std::vector<double> a, std::vector<double>& b)
+{
+  for (size_t i = 0; i < a.size(); ++i)
+  {
+    b[i] = abs(a[i]);
+  }
+}
+
+double mean(std::vector<double> a)
+{
+  double sum = 0.0;
+  for (size_t i = 0; i < a.size(); ++i)
+  {
+    sum += a[i];
+  }
+  if (sum == 0.0)
+  {
+    return 0.0;
+  }
+  return sum / a.size();
+}
+
+InertiaEstimator::InertiaEstimator(double lambda, size_t acc_size)
+{
+  lambda_ = lambda;
+  acc_size_ = acc_size;
+  joint_inertia_ = 0.0;
+
+  // Size the buffers
+  velocity_array_.resize(vel_size_);
+
+  acceleration_array_.resize(acc_size_);
+
+  filtered_acceleration_array_.resize(acc_size_);
+
+  joint_torque_.resize(torque_size_);
+
+  filtered_joint_torque_.resize(fil_tor_size_);
+
+  for (unsigned int i = 0; i < vel_size_; ++i)
+  {
+    velocity_array_[i] = 0.0;
+  }
+
+  for (unsigned int i = 0; i < acc_size_; ++i)
+  {
+    acceleration_array_[i] = 0.0;
+  }
+
+  for (unsigned int i = 0; i < torque_size_; ++i)
+  {
+    joint_torque_[i] = 0.0;
+  }
+
+  // Setup Butterworth filter
+}
+
+InertiaEstimator::InertiaEstimator(hardware_interface::JointHandle joint, double lambda, size_t acc_size)
+{
+  lambda_ = lambda;
+  acc_size_ = acc_size;
+  joint_inertia_ = 0.0;
+
+  // Size the buffers
+  velocity_array_.resize(vel_size_);
+
+  acceleration_array_.resize(acc_size_);
+
+  filtered_acceleration_array_.resize(acc_size_);
+
+  joint_torque_.resize(torque_size_);
+
+  filtered_joint_torque_.resize(fil_tor_size_);
+
+  setJoint(joint);
+
+  for (unsigned int i = 0; i < vel_size_; ++i)
+  {
+    velocity_array_[i] = 0.0;
+  }
+
+  for (unsigned int i = 0; i < acc_size_; ++i)
+  {
+    acceleration_array_[i] = 0.0;
+  }
+
+  for (unsigned int i = 0; i < torque_size_; ++i)
+  {
+    joint_torque_[i] = 0.0;
+  }
+
+  // Setup Butterworth filter
+}
+
+InertiaEstimator::~InertiaEstimator()
+{
+}
+
+double InertiaEstimator::getAcceleration(unsigned int index)
+{
+  return acceleration_array_[index];
+}
+
+void InertiaEstimator::setJoint(hardware_interface::JointHandle joint)
+{
+  joint_ = joint;
+}
+
+void InertiaEstimator::setLambda(double lambda)
+{
+  lambda_ = lambda;
+}
+void InertiaEstimator::setAcc_size(size_t acc_size)
+{
+  acc_size_ = acc_size;
+}
+
+void InertiaEstimator::setNodeHandle(ros::NodeHandle& nh)
+{
+  nh_ = nh;
+}
+
+void InertiaEstimator::configurePublisher(std::string name)
```

Comment:
use const reference to avoid copies
```suggestion
void InertiaEstimator::configurePublisher(const std::string& name)
```